### PR TITLE
docs: improve README examples of `stats/base/dists/negative-binomial`

### LINKED ISSUE #1637: [RFC]: improve README examples of `stats/base/dists/negative-binomial` namespace
### Description

This RFC proposes to improve the [README examples](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/dists/negative-binomial#examples) of the `stats/base/dists/negative-binomial` namespace package. Currently, namespace package READMEs only include minimal examples which are not particularly informative and do not showcase namespace functionality.

Historically, namespace examples were not given much effort due to the rapidly changing nature of the project; however, now that development and organization has settled, it would be good to revisit these packages and add proper examples to better communicate to users how to use namespace contents.

For those wanting to contribute to this issue, you can use the following steps:

- Browse through the [contents](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/dists/negative-binomial) of the `stats/base/dists/negative-binomial` namespace and identify interesting functions and packages to showcase.
- Update the example code block in the "Examples" section of the README for the `stats/base/dists/negative-binomial` namespace and open a PR.
- When updating the README, examples should follow existing project conventions (e.g., ES5).
- Examples may `require` functionality from other packages to help make examples more interesting.

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Userf1995**
Resolves #1637  .

## Description

> What is the purpose of this pull request?

This pull request:

-   Add examples of the functions present in stats/base/dists/negative-binomial namespace and their usage.
-  Adds example demonstration using 7 functions in the README.
-  Adds the same demonstration example in examples/index.js

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1637 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

I've created a real-life use case to make it more interesting, which utilizes some commonly used functions of negative binomial distribution.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

### CODE REVIEW

**Comment by Userf1995** [comment]
@Planeshifter, @Pranavchiku can u guys  review  this PR?

**Review by Pranavchiku — APPROVED** [review]
Looks fine. I am not the best person to review this. At a glance, this looks good.

**Comment by Rejoan-Sardar** [comment]
@Planeshifter @kgryte Anybody please review it,and let me know if any changes need to be made

**Comment by Userf1995** [comment]
@kgryte  please review it,and let me know if any changes need to be made

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md [hunk 1]
```diff
@@ -106,10 +106,43 @@ var y = dist.pmf( 4.0 );
...
+var r = 5.0;
+var p = 1/2;
+
+// Mean can be used to calculate the average number of trials needed to get 5 heads:
+console.log(negativeBinomial.mean(r, p));
+// => 5
+
+// PMF can be used to calculate the probability of getting heads on a specific trial (say on the 8th trial):
+console.log(negativeBinomial.pmf(8, r, p));
+// => ~0.06042480468749999
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md]:
```suggestion
// => ~0.06
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md [hunk 2]
```diff
@@ -106,10 +106,43 @@ var y = dist.pmf( 4.0 );
...
+
+// CDF can be used to calculate the probability up to a certain number of trials (say up to 8 trials):
+console.log(negativeBinomial.cdf(8, r, p));
+// => ~0.8665771484375
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md]:
```suggestion
// => ~0.867
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md [hunk 3]
```diff
@@ -106,10 +106,43 @@ var y = dist.pmf( 4.0 );
...
+
+// Quantile can be used to calculate the number of trials at which you can be 80% confident that the actual number will not exceed:
+console.log(negativeBinomial.quantile(0.8, r, p));
+// => 7
+
+// Standard deviation can be used to calculate the measure of the spread of trials around the mean:
+console.log(negativeBinomial.stdev(r, p));
+// => ~3.1622776601683795
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md]:
```suggestion
// => ~3.162
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md [hunk 4]
```diff
@@ -106,10 +106,43 @@ var y = dist.pmf( 4.0 );
...
+
+// Skewness can be used to calculate the asymmetry of the distribution of trials:
+console.log(negativeBinomial.skewness(r, p));
+// => ~0.9486832980505138
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md]:
```suggestion
// => ~0.949
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md [hunk 5]
```diff
@@ -106,10 +106,43 @@ var y = dist.pmf( 4.0 );
...
+
+// MGF can be used for more advanced statistical analyses and generating moments of the distribution:
+console.log(negativeBinomial.mgf(0.5, r, p));
+// => ~2277.5972997372114
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md]:
```suggestion
// => ~2277.598
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js [hunk 6]
```diff
@@ -18,7 +18,40 @@
...
+var r = 5.0;
+var p = 1/2;
+
+// Mean can be used to calculate the average number of trials needed to get 5 heads:
+console.log(negativeBinomial.mean(r, p));
+// => 5
+
+// PMF can be used to calculate the probability of getting heads on a specific trial (say on the 8th trial):
+console.log(negativeBinomial.pmf(8, r, p));
+// => ~0.06042480468749999
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js]:
```suggestion
// => ~0.06
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js [hunk 7]
```diff
@@ -18,7 +18,40 @@
...
+
+// CDF can be used to calculate the probability up to a certain number of trials (say up to 8 trials):
+console.log(negativeBinomial.cdf(8, r, p));
+// => ~0.8665771484375
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js]:
```suggestion
// => ~0.867
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js [hunk 8]
```diff
@@ -18,7 +18,40 @@
...
+
+// Quantile can be used to calculate the number of trials at which you can be 80% confident that the actual number will not exceed:
+console.log(negativeBinomial.quantile(0.8, r, p));
+// => 7
+
+// Standard deviation can be used to calculate the measure of the spread of trials around the mean:
+console.log(negativeBinomial.stdev(r, p));
+// => ~3.1622776601683795
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js]:
```suggestion
// => ~3.162
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js [hunk 9]
```diff
@@ -18,7 +18,40 @@
...
+
+// Skewness can be used to calculate the asymmetry of the distribution of trials:
+console.log(negativeBinomial.skewness(r, p));
+// => ~0.9486832980505138
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js]:
```suggestion
// => ~0.949
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js [hunk 10]
```diff
@@ -18,7 +18,40 @@
...
+
+// MGF can be used for more advanced statistical analyses and generating moments of the distribution:
+console.log(negativeBinomial.mgf(0.5, r, p));
+// => ~2277.5972997372114
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js]:
```suggestion
// => ~2277.597
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md [hunk 11]
```diff
@@ -106,10 +106,43 @@ var y = dist.pmf( 4.0 );
...
+console.log(negativeBinomial.stdev(r, p));
+// => ~3.162
+
+// Skewness can be used to calculate the asymmetry of the distribution of trials:
+console.log(negativeBinomial.skewness(r, p));
+// => ~0.949
+
+// MGF can be used for more advanced statistical analyses and generating moments of the distribution:
+console.log(negativeBinomial.mgf(0.5, r, p));
+// => ~2277.598
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md]:
```suggestion
console.log( negativeBinomial.mean( r, p ) );
// => 5

// PMF can be used to calculate the probability of getting heads on a specific trial (say on the 8th trial):
console.log( negativeBinomial.pmf( 8, r, p ) );
// => ~0.06

// CDF can be used to calculate the probability up to a certain number of trials (say up to 8 trials):
console.log( negativeBinomial.cdf( 8, r, p ) );
// => ~0.867

// Quantile can be used to calculate the number of trials at which you can be 80% confident that the actual number will not exceed:
console.log( negativeBinomial.quantile( 0.8, r, p ) );
// => 7

// Standard deviation can be used to calculate the measure of the spread of trials around the mean:
console.log( negativeBinomial.stdev( r, p ) );
// => ~3.162

// Skewness can be used to calculate the asymmetry of the distribution of trials:
console.log( negativeBinomial.skewness( r, p ) );
// => ~0.949

// MGF can be used for more advanced statistical analyses and generating moments of the distribution:
console.log( negativeBinomial.mgf( 0.5, r, p ) );
// => ~2277.598
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js [hunk 12]
```diff
@@ -18,7 +18,40 @@
...
+console.log(negativeBinomial.stdev(r, p));
+// => ~3.162
+
+// Skewness can be used to calculate the asymmetry of the distribution of trials:
+console.log(negativeBinomial.skewness(r, p));
+// => ~0.949
+
+// MGF can be used for more advanced statistical analyses and generating moments of the distribution:
+console.log(negativeBinomial.mgf(0.5, r, p));
+// => ~2277.597
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js]:
```suggestion
console.log( negativeBinomial.mean( r, p ) );
// => 5

// PMF can be used to calculate the probability of getting heads on a specific trial (say on the 8th trial):
console.log( negativeBinomial.pmf( 8, r, p ) );
// => ~0.06

// CDF can be used to calculate the probability up to a certain number of trials (say up to 8 trials):
console.log( negativeBinomial.cdf( 8, r, p ) );
// => ~0.867

// Quantile can be used to calculate the number of trials at which you can be 80% confident that the actual number will not exceed:
console.log( negativeBinomial.quantile( 0.8, r, p ) );
// => 7

// Standard deviation can be used to calculate the measure of the spread of trials around the mean:
console.log( negativeBinomial.stdev( r, p ) );
// => ~3.162

// Skewness can be used to calculate the asymmetry of the distribution of trials:
console.log( negativeBinomial.skewness( r, p ) );
// => ~0.949

// MGF can be used for more advanced statistical analyses and generating moments of the distribution:
console.log( negativeBinomial.mgf( 0.5, r, p ) );
// => ~2277.598
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md [hunk 13]
```diff
@@ -106,10 +106,43 @@ var y = dist.pmf( 4.0 );
...
+console.log( negativeBinomial.stdev( r, p ) );
+// => ~3.162
+
+// Skewness can be used to calculate the asymmetry of the distribution of trials:
+console.log( negativeBinomial.skewness( r, p ) );
+// => ~0.949
+
+// MGF can be used for more advanced statistical analyses and generating moments of the distribution:
+console.log( negativeBinomial.mgf( 0.5, r, p ) );
+// => ~2277.598
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/README.md]:
```suggestion
// => ~2277.597
```

File: lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js [hunk 14]
```diff
@@ -18,7 +18,40 @@
...
+console.log( negativeBinomial.stdev( r, p ) );
+// => ~3.162
+
+// Skewness can be used to calculate the asymmetry of the distribution of trials:
+console.log( negativeBinomial.skewness( r, p ) );
+// => ~0.949
+
+// MGF can be used for more advanced statistical analyses and generating moments of the distribution:
+console.log( negativeBinomial.mgf( 0.5, r, p ) );
+// => ~2277.598
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/negative-binomial/examples/index.js]:
```suggestion
// => ~2277.597
```

**Review by Planeshifter — APPROVED** [review]
Thanks! Example looks good and is a definite improvement; I added spaces around function parentheses and rounded console.log annotations to three digits after decimal point.

Will merge once CI passes.

**Pull request merged**